jwst/assign_wcs/util.py

```diff
@@ -822,10 +829,35 @@ def _create_grism_bbox(input_model, mmag_extract=99.0,
                                                      sky_bbox_ur=obj.sky_bbox_ur,
                                                      xcentroid=xcenter,
                                                      ycentroid=ycenter,
-                                                     is_extended=obj.is_extended))
-    if len(grism_objects) == 0:
-        log.warning("No grism objects saved, check catalog")
-    return grism_objects
+                                                     is_extended=obj.is_extended,
+                                                     isophotal_abmag=obj.isophotal_abmag))
+
+    # At this point we have a list of grism objects limited to
+    # isophotal_abmag < mmag_extract. We now need to further restrict
+    # the list to the N brightest objects, as given by nbright.
+    if nbright is None:
+        # Include all objects, regardless of brightness
+        final_objects = grism_objects
+    else:
+        # grism_objects is a list of objects, so it's not easy or practical
+        # to sort it directly. So create a list of the isophotal_abmags, which
+        # we'll then use to find the N brightest objects.
+        mags = []
+        for obj in grism_objects:
+            mags.append(obj.isophotal_abmag)
+        indxs = np.argsort(mags)
+
+        # Create a final grism object list containing only the N brightest objects
+        final_objects = []
+        for i in list(indxs[:nbright]):
+            final_objects.append(grism_objects[i])
+        del grism_objects
```

Comment:
One way to collapse 845 - 853 
```
indxs = np.argsort([obj.isophotal_abmag for obj in grism_objects])
final_objects = []
final_objects = [grism_objects[i] for i in indxs[:nbright]]
```

Reply:
It's possible also to sort the original table in get_object_info

```
catalog =QTable.read(catalog_file, format='ascii.ecsv')
catalog.sort('isophotal_abmag')
```

However, this appears to be much slower than working with lists.

Reply:
> One way to collapse 845 - 853
> 
> ```
> indxs = np.argsort([obj.isophotal_abmag for obj in grism_objects])
> final_objects = []
> final_objects = [grism_objects[i] for i in indxs[:nbright]]
> ```

Implemented in latest commit. Thanks!